pkg/tasks/tasks.go

```diff
@@ -39,23 +41,23 @@ func NewTaskRunner(client *client.Client, taskGroups ...*TaskGroup) *TaskRunner
 
 // RunAll executes all registered task groups sequentially. For each group the
 // taskGroup.RunConcurrently function is called.
-func (tl *TaskRunner) RunAll() (string, error) {
+func (tl *TaskRunner) RunAll() TaskGroupErrors {
 	for i, tGroup := range tl.taskGroups {
 		klog.V(2).Infof("processing task group %d of %d", i+1, len(tl.taskGroups))
-
-		if name, err := tGroup.RunConcurrently(); err != nil {
-			return name, err
+		tErrors := tGroup.RunConcurrently()
+		if len(tErrors) > 0 {
+			return tErrors
```

Comment:
Mostly something I'm wondering about: if `tGroup.RunConcurrently()` returned `*TaskGroupErrors` we could forego the error checking:
```
return tGroup.RunConcurrently()
```

And when we want to check, we can rely on the idiomatic `if err != nil`.
I might be missing something here, hence consider this rather a question from me :)

Reply:
This is done because the second task group will not run if we do a 
return tGroup.RunConcurrently()

Reply:
Fair point, though the question still stands about the more idiomatic error checking. In `RunAll` we'd then have
```
if tError != nil {
        return tError
}
```

Reply:
On re-reading, this doesn't even need a change in return type. Since `TaskGroupErrors` is a slice, checking for nil should just work. I realize this is a nit, but idiomatic error checking would help readability here I think.

Reply:
Since it is an array of errors, I would expect the caller to check for either `nil` or `len() > 0`. Although it is safer to check the len since an empty slice may not be nil. I would agree with Jan here and say that we should directly return the array of errors as is.

Reply:
The length check is added only to decide if loop should continue, if there are errors return the errors and and if there are none continue the loop.

Reply:
You're right, simply returning the slice won't work. Apologies, we've discussed this before.

What do you think about checking against `nil` though. I think that would help readability. @dgrisonnet argument is certainly valid (an empty slice can also not be equal to `nil` if it has a capacity) though I think one can argue that we'll not see those slices here.

Reply:
Ah indeed I didn't have a throughout look at the code surrounding this check my bad. I think what @prashbnair did is correct, and I would continue to argue against checking for nil.
Since we know that `TaskGroupErrors` is an array of errors, the best to check if an error occurred is to check that its length is superior to 0. Taking it literally, checking for nil would mean: `is this array allocated?` whereas checking its length would translate to `are there any errors in this array?`.

Reply:
ack, thank you.